tests/test_github.py

```diff
@@ -248,9 +248,11 @@ def test_report_pull_status(self):
         self.waitUntilSettled()
         # We should have a status container for the head sha
         self.assertIn(A.head_sha, A.statuses.keys())
-        # We should only have one status for the head sha
-        self.assertEqual(1, len(A.statuses[A.head_sha]))
-        check_status = A.statuses[A.head_sha][0]
+        # We should only have one status for the head sha but adding a
+        # second pipeline with the same trigger but differen project
+        # currently results in both pipelines reporting status
```

Comment:
There were problems in the reporting structure, recently fixed in v3, so that events are not reported cross-connection. I don't think we will be fixing v2 at all.